src/services/apiService.ts

```diff
@@ -0,0 +1,295 @@
+/**
+ * API Service
+ *
+ * This service provides a mock API interface for development using fixture data.
+ * Designed to be easily replaceable with real API endpoints when backend is ready.
+ *
+ * Usage examples:
+ * ```typescript
+ * // Get cases with sorting
+ * const cases = await getCases('new', 'desc');
+ * 
+ * // Direct API service usage
+ * const response = await apiService.getCases({ caseType: 'new', sortOrder: 'desc' });
+ * ```
+ */
+
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { setTimeout as delay } from 'timers/promises';
+import type { CaseData, DateOfBirth } from '#types/case-types.js';
+import {
+  isValidDateOfBirth,
+  safeString,
+  safeOptionalString,
+  isRecord,
+  devLog,
+  devError
+} from '#src/scripts/helpers/index.js';
+
+// Constants
+const DEFAULT_PAGE = 1;
+const DEFAULT_LIMIT = parseInt(process.env.PAGINATION_LIMIT ?? '20', 10); // Configurable via env
+const MOCK_DELAY = 50;
+const ZERO_TOTAL = 0;
+const MIN_PAGE_VALUE = 1;
+
+/**
+ * API response structure that matches expected backend format
+ */
+export interface ApiResponse<T> {
+  data: T[];
+  meta: {
+    total: number;
+    page: number;
+    limit: number;
+    sortBy: string;
+    sortOrder: 'asc' | 'desc';
+  };
+  status: 'success' | 'error';
+  message?: string;
+}
+
+/**
+ * API request parameters for cases
+ */
+export interface CaseApiParams {
+  caseType: 'new' | 'accepted' | 'opened' | 'closed';
+  sortBy?: string;
+  sortOrder?: 'asc' | 'desc';
+  page?: number;
+  limit?: number;
+  filters?: Record<string, unknown>;
+}
+
+/**
+ * Transform dateOfBirth object to string format (DD MMM YYYY)
+ * @param {DateOfBirth} dateOfBirth DateOfBirth object
+ * @returns {string} Formatted date string
+ */
+function transformDateOfBirth(dateOfBirth: DateOfBirth): string {
+  const { day, month, year } = dateOfBirth;
+  return `${day} ${month} ${year}`;
+}
+
+/**
+ * Transform raw case item to display format
+ * @param {unknown} item Raw case item
+ * @returns {CaseData} Transformed case item
+ */
+function transformCaseItem(item: unknown): CaseData {
+  if (!isRecord(item)) {
+    throw new Error('Invalid case item: expected object');
+  }
+
+  const { dateOfBirth: dateOfBirthValue } = item;
+
+  return {
+    fullName: safeString(item.fullName),
+    caseReference: safeString(item.caseReference),
+    refCode: safeString(item.refCode),
+    dateReceived: safeString(item.dateReceived),
+    caseStatus: safeString(item.caseStatus),
+    dateOfBirth: isValidDateOfBirth(dateOfBirthValue) ? transformDateOfBirth(dateOfBirthValue) : '',
```

Comment:
Do we want to return anything else other that `''` when it's not a valid DOB? DOB isn't optional so should we have a warning of some sort?

Reply:
This applies only to mock data. Once we connect to the Case Details API, all data will be valid, and none of these adjustments will be necessary. This is simply to ensure that our local JSON file follows the correct structure for development purposes.